caddyhttp/gzip/setup.go

```diff
@@ -16,6 +19,8 @@ func setup(c *caddy.Controller) error {
 		return err
 	}
 
+	initWriterPool()
```

Comment:
`setup` can be called multiple times, even while a server is running, for example when it gets reloaded. This will run while the server is already up and running. Either the writer pool map should be guarded with a RWMutex -OR- it should only be initialized once, like in `init()` instead of `setup`.